feature/flags.go

```diff
@@ -46,3 +46,6 @@ const PersistentStorage = "persistent-storage"
 // StrictMigration will cause migration to error if there are unexported
 // values for annotations, status, status history, or settings.
 const StrictMigration = "strict-migration"
+
+// CAASModel enables creating models on CAAS infrastructure (k8s, etc)
+const CAASModel = "caas-model"
```

Comment:
My thought was that we may want a separate CAAS feature flag for future CAAS-related things, such as bootstrapping a controller into a pre-existing CAAS.
However I don't feel that strongly about it - if that's too much flag proliferation, then I'm fine with just "CAAS".